ParameterDataApi.yaml

```diff
@@ -0,0 +1,226 @@
+      - ParameterDataDefinitionsAuth:
+        - fraihmwork.parameter.data.write
+
+components:
+  schemas:
+    Response:
+      type: object
+      description: A default uuid / message combination to convey simple success / error messages to the client
+      properties:
+        itemId:
+          type: string
+          format: uuid
+        message:
+          type: string
+          description: The success message provided back to the client
+      required:
+        - message
+        
+    ColumnDefinition:
+      type: array
+      items:
+        $ref: '#/components/schemas/ColumnElementDefinition'
+          
+    ColumnElementDefinition:
+      type: object
+      description: The column elements      
+      properties:
+        
+        name: 
+          description: The name of the column
+          type: string
+          pattern: '^[a-zA-Z0-9]{2,}'
+          
+        dataType:
+          $ref: '#/components/schemas/DataType'
+                   
+      required:
+        - name
+        - dataType
+
+    DataDefinition:
+      type: array
+      items: 
+        $ref: '#/components/schemas/DataElement'
+        
+    DataElement:
+      type: object
+      properties:
+        timestamp:
+          description: the timestamp of the sample
+          type: string
+          format: date-time
+            
+        fields:
+          type: array
+          items:
+            $ref: '#/components/schemas/ValueSpec'
+            
+      required:
+        - timestamp
+        - fields
+
+    DataType:
+      type: string
+      enum:
+        - NUMBER
+        - INTEGER
+        - STRING
+        
+    HeaderDefinition:
+      type: object
+      properties: 
+        startTime:
+          description: The start time for the time series
+          type: string
+          format: date-time
+        
+        endTime: 
+          description: The end time for the time series
+          type: string
+          format: date-time
+        
+        recordCount: 
+          description: The number of records in the data.
+          type: integer
+          
+        columns:
+          $ref: '#/components/schemas/ColumnDefinition'
+        
+      required:
+        - startTime
+        - endTime
+        - recordCount
+        - columns
+
+    RenderType:
+      type: string
+      enum:
+        - VALUE
+
+    TimeSeriesDataDefinition:
+      type: object
+      description: A definition of time series data. This is loosely based upon Json Time Series, Version 1.0. (https://docs.eagle.io/en/latest/reference/index.html)
+      properties:
+        docType:
+          type: string          
+          description: The type of the document
+
+        version:
+          type: string
+          description: The version string for the time series data definition
+
+        header:
+          $ref: '#/components/schemas/HeaderDefinition'
+
+        data:
+          $ref: '#/components/schemas/DataDefinition'
+
+      required: 
```

Comment:
According to the JSON Time Series documentation, docType and version should be required fields.

Reply:
We're deviating from JTS quite heavily, now. We can keep with the structure of the specification and allow for the fields to be present, but I don't think we should require these fields.